---
title: "A Primer on Molecular Docking (Part 1): Structure & Binding"
description: "How to evaluate protein structures and understand molecular binding for better docking results."
date: 2025-09-30
isPublished: true
tags: ["Long Reads", "Biophysics", "Biology", "Chemistry"]
heroImage: '/images/posts/pexels-googledeepmind-17497303.jpg'
heroImageCaption: 'Photo by Google Deepmind'
---
import { YouTube } from 'astro-embed';
import BeforeAfterSlider from '../../components/BeforeAfterSlider.tsx';

Molecular docking is often presented with a touch of magic for predicting the binding affinity between a ligand and a receptor. However, having spent some time in my undergraduate thesis on protein-ligand interactions, I've found this perception to be largely misleading. While docking is undeniably a fascinating technique, the reality of docking is much more nuanced.

Traditional experimental techniques such as isothermal titration calorimetry (ITC) or surface plasmon resonance (SPR) can reliably measure binding affinity in-lab, but they're relatively slow and expensive. Docking allows you to screen thousands if not millions of potential leads in silico before you ever step into a lab.

Yet, it's not a magic bullet. It's a fantastic tool, but it also operates on a foundation of assumptions and requires careful preparation to yield meaningful results.  It serves as a crude estimator, incredibly useful as long as you know what you're doing.

This series aims to explore the process over three focused parts:
- **Part 1**: The fundamentals of how molecules bind.
- **Part 2**: A look under the hood at how docking works.
- **Part 3**: A practical guide to doing it right.

This series is intended for anyone with some background in molecular biology and biochemistry. Some of the topics that I will assume you're familiar with are: protein expression, molecular cloning, and some basic on chemical bond interactions and protein structure. Whether you're a curious newcomer or you've already run a few simulations, I promise there will be something interesting here for you.

---

# Introduction

Before we simulate any docking experiments, two things must be in place: a solid protein structure and a good understanding of what “binding” actually means physically. First we walk through X-ray crystallography from crystal to PDB entry, highlighting the quality metrics, ranging from resolution, B-factors, R-factor, steric clashes, to Ramachandran plot, occupancy, and more.

We then will explore the thermodynamics of binding, translating the experimental $\Delta G$ into the same force-field components that docking programs estimate: van der Waals interactions, Coulombic interactions, desolvation penalties and the entropy lost upon locking flexible groups. Seeing how these terms actually contribute to the binding energy will let you better judge docking poses and scores.

---

# Proteins and the magic of determining their structure

Proteins are the workhorses of biology, acting as enzymes, receptors, transporters, and structural components that execute various cellular functions. Many of us probably have seen the illustration of a protein 3D structure somewhere. But, have you ever thought, **how can we know their structure?**

There are many ways we can determine the structure of a given protein, such as x-ray crystallography, nuclear magnetic resonance (NMR), and recently, cryo-EM<a href="#footnote-ref-1" id="footnote-1" className="footnote-ref" aria-describedby="footnote-label" title="Jump to footnote">[1]</a>. As the oldest method amongst the three, **x-ray crystallography** is the dominant method for determining protein structures. It has tons of drawbacks, sure, but it's the most established method. X-ray crystallography has been the mainstay historically and still represents the majority of PDB entries, though cryo-EM's share has grown rapidly for large assemblies<a href="#footnote-ref-2" id="footnote-2" className="footnote-ref" aria-describedby="footnote-label" title="Jump to footnote">[2]</a>. As such, we will talk more about it and put aside the other two methods.

It may be tempting to dismiss this section as too theoretical, after all, we can just download the structure from RCSB. However, the goal of this section is not to teach you on how to do it, but to show the assumptions, limitations, and shortcuts that were made to even have proteins structure. We are not talking about docking yet, the structure determination itself is full of flaws. Every structure, every entry on PDB will have some flaw, either by the experiment or the crystallographer. After all, they're models, a projection of reality. By knowing those, you can be more skeptical and better manage your expectations on docking results later. 

## X-ray crystallography crash course

Displayed below is the general workflow of x-ray crystallography (see Figure 1). It seems straightforward at first glance: crystallize the protein, shoot it with X-rays, and solve the structure. While the physics and computation can be complex, the biological part seems simple enough. However, you can't be more wrong. I will explain briefly some of the major challenges of expressing and crystallising the protein, then the complexity of the modelling and refining the model afterwards.

<div style="text-align: center;  margin: 0 auto; width: 90%">
  ![General workflow of x-ray crystallography. ](/images/posts/xray-crystallography.jpg)
  <p style="font-size: 0.8em; margin-top: -2.5rem;"><b>Figure 1</b>: General workflow of x-ray crystallography. <br/> [By Jonathan Gutow, CC BY 4.0, <a href="https://chem.libretexts.org/Courses/University_of_Wisconsin_Oshkosh/Chem_370%3A_Physical_Chemistry_1_-_Thermodynamics_(Gutow)/10%3A_Aggregates_and_Macromolecules/10.02%3A_X-Ray_Crystallography">Chem Libretext</a>]</p>
</div>


### The struggle of crystallisation

Before we can even begin, we need to produce a large quantities of pure and stable samples of our protein of interest. Most proteins are simply not abundant enough in their native sources, hence we often resort to **heterologous expression**, that is expressing a protein outside of its native organism; most frequently one of the many designd *E. coli* strains like BL21, yeasts, or even mammalian expression system such as Human Embryonic Kidney (HEK). This is where the problem begins.

Different organisms have different cellular internal environments and molecular machinery. For example, different organisms have different codon preferences, hence we need to optimize the codons for our genetic code. Thankfully, modern bioinformatics has made this hardly a problem since codon optimizers are widely available.

A more pressing issue is that of post-translational modifications (PTMs). Prokaryotes generally lack PTMs, unlike eukaryotes. While the absence of PTMs can be an advantage, as we can avoid the problem of conformationally heterogeneous glycosylation, the lack of PTMs can still render the protein nonfunctional. However, if the protein's fold is not compromised, the structure can still provide insights, albeit with caveats. Secondly, protein expression induces metabolic stress in the host. If not handled properly, this can cause misfolding, resulting in inclusion bodies. Heterologous organisms also don't always have the necessary chaperones to fold the protein correctly. Last but not least, disulfide bonds (Cys-S-S-Cys) require the chemical oxidation of two adjacent cysteines to form. Unfortunately, a bacterium's intracellular environment is generally reducing, which means complex disulfide bond formation in the cytoplasm sometimes fails to occur properly. This problem can be alleviated by using yeast or other eukaryotic expression systems, but specialized E. coli strains that have mutations in the trxB and gor genes can also be used.

We won't talk too much about fusions and tags since they're beyond the scope of this post, but you need to remember that these are also needed for purification, adding another factor to our already complex situation.

<YouTube id="1aeemIQJ5YE" title="How do proteins form crystals?"/>

Next, crystallisation. What is a crystal anyway? Crystals are basically just an array of highly ordered molecules. Table salt is a common example of crystal. However, unlike table salt (NaCl) which has a strong bond thanks to their strong ionic bond, proteins don't get to enjoy that kind of luxury. Protein crystals are largely kept together by weaker forces between side-chains, such as vdW, hydrogen bond, and so on. As for the crystallisation process itself, basically we tries to nudge the proteins to a state of supersaturation by slowly removing solvents. However, that process, more often than not, fails miserably. We need to repeat our experiments multiple times, using different solvents, additives, concentration, pH, temperature, and so on until it's just right for the proteins to crystallise (most commonly using vapor diffusion methods like hanging drop). Some proteins can't even be crystallised, at all. Some *suggests* that the reason why proteins don't crystallise well is perhaps due to evolutionary reason: you don't want your proteins to clump together in the nature. 

<div style="text-align: center;  margin: 0 auto; width: 90%">
  ![ Salt crystals (NaCl)](/images/posts/sa_fig01.png)
  <p style="font-size: 0.8em; margin-top: -2.5rem;"><b>Figure 2</b>: Salt crystals (NaCl). <br/> [By Japan Aerospace Exploration Agency (JAXA), © JAXA, <a href="https://humans-in-space.jaxa.jp/protein/en/public/about/structural_analysis.html">Humans in Space</a>]</p>
</div>

<div style="text-align: center;  margin: 0 auto; width: 90%">
  ![ Protein crystals (haemoglobin)](/images/posts/sa_fig02.png)
  <p style="font-size: 0.8em; margin-top: -2.5rem;"><b>Figure 3</b>: Protein crystals (haemoglobin).<br/> [By Japan Aerospace Exploration Agency (JAXA), © JAXA, <a href="https://humans-in-space.jaxa.jp/protein/en/public/about/structural_analysis.html">Humans in Space</a>]</p>
</div>

A good crystal should be well ordered with minimal deformities, otherwise the resulting diffraction pattern can be poor and unusable for downstream processing. Hence, merely having a crystal is not enough. Methods such as Surface Entropy Reduction (SER) is sometimes used to reduce entropy on the surface. The idea is to identify surface residues with long, flexible side chains, such as lysine and to replace them with small, more rigid residue, such as alanine. By mutating those floppy surface side chains, the entropic penalty for crystallisation can be minimised, making the process easier to achieve. One might ask: "Wont that change the conformation?" Well, yeah probably, but we can try to use computational model to minimise the potential effect, especially near critical regions like binding site. After all, a slightly wrong model is better than having no model at all, right?

<div style="text-align: center;  margin: 0 auto; width: 90%">
  ![ Protein crystals (haemoglobin)](/images/posts/sa_fig04.png)
  <p style="font-size: 0.8em; margin-top: -2.5rem;"><b>Figure 4</b>: Illustration of well ordered crystal.<br/> [By Japan Aerospace Exploration Agency (JAXA), © JAXA, <a href="https://humans-in-space.jaxa.jp/protein/en/public/about/structural_analysis.html">Humans in Space</a>]</p>
</div>

To illustrate how crystallisation can introduce problems, consider the case of HIV-1 integrase. In its first crystal structure (PDB: 1itg), the protein was crystallised in the presence of cacodylate <a href="#footnote-ref-8" id="footnote-8" className="footnote-ref" aria-describedby="footnote-label" title="Jump to footnote">[8]</a>. That cacodylate happened to attach itself covalently to a cysteine in the enzyme's active site. Because of this attachment, the key catalytic residues—Asp64, Asp116, and Glu152—ended up arranged in the wrong geometry. The structure looks valid crystallographically, but it wasn't biologically accurate since it made the enzyme looked inactive. It only became clear that the catalytic residues had been “knocked out of alignment” when later structures from HIV-1 integrase crystals grown without cacodylate were solved. In those, the enzyme appeared in its proper functional geometry <a href="#footnote-ref-8" id="footnote-8" className="footnote-ref" aria-describedby="footnote-label" title="Jump to footnote">[9]</a>. 

<div style="text-align: center;  margin: 0 auto; width: 90%">
  ![ Lysozyme crystals observed through polarizing filter in Nikon SMZ800 microscope.](/images/posts/960px-Lüsosüümi_kristallid_pv.jpg)
  <p style="font-size: 0.8em; margin-top: -2.5rem;"><b>Figure 5</b>: Lysozyme crystals observed through polarizing filter in Nikon SMZ800 microscope. <br/> [By Taavi Ivan, CC BY-SA 3.0, <a href="https://commons.wikimedia.org/w/index.php?curid=22143196">Wikimedia</a>]</p>
</div>


If somehow, by divine benevolence, you manage to crystallise your protein interest, you will get pretty looking crystals, like the one displayed in Figure 2. One interesting property of protein crystals are that they behave different optically, hence **we can use polarising lens to differentiate our crystal from salt crystals**. Once we have our crystals prepared, we can do the diffraction experiment by shining x-rays to our crystals, and recording the diffraction pattern.

### Modelling and refinement

<div style="text-align: center;  margin: 0 auto; width: 90%">
  ![General workflow of x-ray crystallography. ](/images/posts/fft_amp_phase_swap.png)
  <p style="font-size: 0.8em; margin-top: -2.5rem;"><b>Figure 1</b>: General workflow of x-ray crystallography. <br/> [By Rayhan M, CC BY 4.0]</p>
</div>

<div style="text-align: center;  margin: 0 auto; width: 90%">
  ![General workflow of x-ray crystallography. ](/images/posts/cat_amp_only.png)
  <p style="font-size: 0.8em; margin-top: -2.5rem;"><b>Figure 1</b>: General workflow of x-ray crystallography. <br/> [By Rayhan M, CC BY 4.0]</p>
</div>

Having done our diffraction experiment, the result can then be processed with computer programs using Fourier transform. Said process will produce an electron density map, often called Structure Factor (SF). It contains the processed data of the electron density map. Basically, we can "see" how the electrons of our protein structure is mapped in 3D space. So, we don't directly get the residues or even the atoms, only the distribution of the electrons. One such example can be seen on Figure 3 as visualised with COOT.

<div style="text-align: center;  margin: 0 auto; width: 90%">
  ![Illustration of the electron density map of 6GLP, visualised with COOT.](/images/posts/electron_density_map.jpg)
  <p style="font-size: 0.8em; margin-top: -2.5rem;"><b>Figure 6</b>: Illustration of the electron density map of 6GLP, visualised with COOT.<br/> [By Rayhan M, CC BY 4.0]</p>
</div>

Afterwards, the job of deciding what residue belongs to each site falls upon the crystallographer. They're tasked with "guessing" which residue is producing said electron density map. Of course, it's not totally random, we can make educated guesses by the shape and length of the blob. For example, in Figure 4, we can guess that it's a Tyr since it has a visible shape of a phenol ring.

<div style="text-align: center;  margin: 0 auto; width: 90%">
  ![Illustration of the electron density map of 6GLP, visualised with COOT.](/images/posts/edm_mapping.jpg)
  <p style="font-size: 0.8em; margin-top: -2.5rem;"><b>Figure 7</b>: Illustration of the electron density map of 6GLP, visualised with COOT. <br/> [By Rayhan M, CC BY 4.0]</p>
</div>

Since we are modelling based on the electron density map, the resolution of the EDM matters a lot. To see why, let's take a look at Figure 12. As the resolution decreases from 1.0 to 6.0 $\AA$, we can see that the EDM gets more and more fuzzy, and the detail of the structure gets lost. We may still be able to see the general shape of the protein, but the details are lost. This is why we care about the resolution of the EDM, because low resolution EDM forces us to make more assumptions regarding the exact positions of the atoms.

<div style="text-align: center; margin: 2rem auto; width: 90%">
  <video 
    width="100%" 
    height="auto" 
    autoplay 
    loop 
    muted 
    style="max-width: 600px; border-radius: 8px;"
  >
    <source src="/images/posts/resolution_grow.webm" type="video/webm" />
    Your browser does not support the video tag.
  </video>
  <p style="font-size: 0.8em; margin-top: 0.5rem;"><b>Figure 12</b>: EDM comparison of various resolutions from 1.0 to 6.0 $\AA$, visualised with ChimeraX. <br/> [By Rayhan M, CC BY 4.0]</p>
</div>

It's also important to note that most protein structure data don't include hydrogen atoms, due to their size. Remember that hydrogen only has 1 electron, as opposed to other atoms such as carbon, which has 4 electrons, or oxygen, which has 6 electrons. Therefore, the electron density of hydrogen is very low, making it hard to resolve with X-ray crystallography. Sub-angstrom resolution has been achieved with various techniques, hence allowing us to resolve hydrogen atoms, but it's still relatively rare. For this reason, we usually need to add hydrogen atoms to our model explicitly. This technique is called **protonation**.

Protonation is a complex process, and it's not always straightforward to determine the correct protonation state of a residue. For example, some residues are known to have a wide range of pKa, such as histidine. 

## Assessing model quality

Since we have established the experiments and modelling process are full of flaws, we need to measure how good exactly our model is. There are many ways to quantify that, which we will discuss shortly.

### Resolution

Crystallographic resolution is an important measure of the quality and detail in a structure determined by X-ray diffraction<a href="#footnote-ref-3" id="footnote-3" className="footnote-ref" aria-describedby="footnote-label" title="Jump to footnote">[3]</a>. At its core, resolution in crystallography is a measure of the smallest distance at which two individual features can be distinguished in the calculated electron density map. A "high-resolution" structure, indicated by a low value (e.g., 1.5 $\AA$), means that the atomic positions are well-defined and individual atoms can be clearly observed. Conversely, a "low-resolution" structure, represented by a higher value (e.g., 3.5 $\AA$), will only show the general outline of the molecular backbone.

**However, it's misleading to think of this resolution as being uniform across the entire molecule.** Instead, it represents an overall average, while different parts of the structure can be more or less ordered.

<div style="text-align: center;  margin: 0 auto; width: 90%">
  ![Illustration of the electron density map of 2AXW, visualised with COOT.](/images/posts/resolution_plot.png)
  <p style="font-size: 0.8em; margin-top: -2.5rem;"><b>Figure 9</b>: Electron density peak resolution, plot inspired by Benhard Rupp (Biomolecular Crystallography 2010 pp. 453). <br/> [By Rayhan M, CC BY 4.0]</p>
</div>

Another common misconception is that the resolution stated in crystallographic table ($d_{min}$) represents the limit of resolution (LoR). However, that is only a conservative approximation. The actual limit of resolution is determined by the electron density peak resolution, as shown in Figure 9.

### B-factor

B-factor is a value assigned to each atom in the crystal structure that quantifies its degree of thermal motion or static disorder<a href="#footnote-ref-4" id="footnote-4" className="footnote-ref" aria-describedby="footnote-label" title="Jump to footnote">[4]</a>. In simpler terms, it measures how much an atom "wobbles" or deviates from its average position within the crystal lattice. A low B-factor means an atom is well-ordered and its position is precisely defined in the EDM. These are typically atoms in the core of a protein. In contrast, a high B-factor indicates that an atom is more mobile or exists in multiple conformations, which is common for flexible loops.

**Regions of a protein with high B-factors inherently correspond to areas of lower local resolution.** In these flexible regions, the electron density is smeared out, making the precise placement of atoms more ambiguous. Conversely, parts of the structure with low B-factors are well-ordered and contribute to the high-resolution features of the electron density map. Therefore, while a single resolution value is often assigned to an entire structure, the B-factors reveal the underlying heterogeneity. Moreover, if an atom wobbles too much, a crystallographer might even mark the atom or even the whole residue as missing, which you should fix before docking.

There are two possible causes for this artifact. The first one is called **static disorder**.  Since our diffraction experiment uses protein crystals, which in itself consists of many proteins, some differences are bound to happen. Not every protein will fold in the exact same way. In practice, we usually average them out, by this can cause the electron density map to be "fuzzy" if the conformations vary wildly from unit cells to the others. Think of it like telling a bunch of people to strike a peace pose, it's bound to have some variance since every person might interpret it differently.

Secondly, there's this thing called **dynamic motion**. As we all know, proteins are not static, atoms are not either. Even when being crystallised, some random motions at the atomic level is still bound to happen, particularly among places where they're not too rigid (e.g., side-chains). Think about this one like capturing people standing still. Even if we assume that they're able to stand perfectly still, some little movements are still inevitable.

<div style="text-align: center; margin: 2rem auto; width: 90%">
  <video 
    width="100%" 
    height="auto" 
    autoplay 
    loop 
    muted 
    style="max-width: 600px; border-radius: 8px;"
  >
    <source src="/images/posts/bfactor-spin.webm" type="video/webm" />
    Your browser does not support the video tag.
  </video>
  <p style="font-size: 0.8em; margin-top: 0.5rem;"><b>Figure 12</b>: Visualization of B-factors in a protein structure (PDB ID: 2LYZ). Red regions indicate high B-factors, blue regions show low B-factors, visualised with ChimeraX. <br/> [By Rayhan M, CC BY 4.0]</p>
</div>

<BeforeAfterSlider 
  beforeImage="/images/posts/pexels-nguyendesigner-12392816.jpg"
  afterImage="/images/posts/pexels-nguyendesigner-12392816_edited.jpg"
  beforeAlt="Original phi dihedral angle rotation example"
  afterAlt="Edited phi dihedral angle rotation example"
  caption="Real life illustration of b-factor, blue indicating low b-factor, while red indicating high b-factor. Slide from A to B to see the difference. <br/> [Original photo by Pew Nguyen, CC BY 4.0, <a href='https://www.pexels.com/photo/female-vendor-carrying-products-12392816/'>Pexel</a>]"
  figureNumber="Figure 13"
  width="70%"
  client:load
/>


### Occupancy

<div style="text-align: center;  margin: 0 auto; width: 90%">
  ![Illustration of the electron density map of 2AXW, visualised with COOT.](/images/posts/2axw_example.jpg)
  <p style="font-size: 0.8em; margin-top: -2.5rem;"><b>Figure 9</b>: Alternative conformation of Arg63 in 2AXW, visualised in RCSB. Highlighted with pink overlay is the side-chain alternative conformation. <br/> [By Rayhan M, CC BY 4.0]</p>
</div>

People often confuse between occupancy and B-factor. They're related, but fundamentally different. Occupancy is the percentage of samples that contain a particular atom. If it's 1.00, it means the atom is present in all samples. If it's 0.50, it means the atom is present in half of the samples.

However, in practice, it's not always so simple. The most common reason is that the atom is present in multiple samples, but not all samples. This is often seen in the case of side-chain alternative conformations. For example, in Figure 6, Arg63 has two alternative conformations, one is the "normal" one, and the other is the "flipped" one. The "flipped" one is present in half of the samples. Therefore, the occupancy of Arg63 is 0.50 for each conformation.

Unfortunately, sometimes there aren't enough data to support the existence of a different conformation. In this case, the occupancy is 1.00 for all samples, but the crystallographer will assign a high B-factor to the atom to indicate that it's not well-ordered (since it appears to be present in different places, but doesn't justify a different conformation). At the end of the day, crystallography is not an exact science, and we need to make the best guess with the data we have. When docking, you should also consider the occupancy of the atoms and to pick one of the alternative conformations.

You can see below how the different conformations are represented in the PDB file of 2AXW for Arg63, specifically starting from line 993.

```shell
# Snippet from the PDB file of 2AXW for Arg63
ATOM    989  N   ARG A  63       6.993  48.721  17.381  1.00  7.41           N  
ANISOU  989  N   ARG A  63     1121    819    875    -66      5    -18       N  
ATOM    990  CA  ARG A  63       7.475  47.391  17.007  1.00  7.44           C  
ANISOU  990  CA  ARG A  63     1040    877    910   -124     36     26       C  
ATOM    991  C   ARG A  63       6.307  46.427  16.943  1.00  7.26           C  
ANISOU  991  C   ARG A  63     1010    899    848    -47     80      3       C  
ATOM    992  O   ARG A  63       5.514  46.344  17.887  1.00  7.45           O  
ANISOU  992  O   ARG A  63      984    932    912    -91     49   -112       O  
ATOM    993  CB AARG A  63       8.489  46.932  18.030  0.50  8.16           C  
ANISOU  993  CB AARG A  63      952   1062   1085   -114    -54     68       C  
ATOM    994  CB BARG A  63       8.501  46.882  18.008  0.50  8.71           C  
ANISOU  994  CB BARG A  63     1066   1088   1153    -76    -93     67       C  
ATOM    995  CG AARG A  63       9.361  45.817  17.673  0.50  9.04           C  
ANISOU  995  CG AARG A  63     1173    987   1272    -76    130     41       C  
ATOM    996  CG BARG A  63       9.857  47.509  17.828  0.50 11.24           C  
ANISOU  996  CG BARG A  63     1327   1429   1513   -218   -192     30       C  
ATOM    997  CD AARG A  63      10.169  45.309  18.897  0.50 11.98           C  
ANISOU  997  CD AARG A  63     1297   1687   1568     14   -148    -37       C  
ATOM    998  CD BARG A  63      10.981  46.795  18.527  0.50 14.43           C  
ANISOU  998  CD BARG A  63     1610   1897   1976    -14   -119     24       C  
ATOM    999  NE AARG A  63      11.112  46.323  19.377  0.50 15.40           N  
ANISOU  999  NE AARG A  63     1964   1917   1970    -50    -12   -145       N  
ATOM   1000  NE BARG A  63      11.146  45.427  18.062  0.50 16.21           N  
ANISOU 1000  NE BARG A  63     2214   1949   1994     19    -59    -29       N  
ATOM   1001  CZ AARG A  63      12.437  46.270  19.217  0.50 19.07           C  
ANISOU 1001  CZ AARG A  63     2174   2548   2521    -37     92    -75       C  
ATOM   1002  CZ BARG A  63      12.193  44.636  18.320  0.50 17.96           C  
ANISOU 1002  CZ BARG A  63     2274   2328   2222     43     48     -6       C  
ATOM   1003  NH1AARG A  63      13.018  45.251  18.586  0.50 19.99           N  
ANISOU 1003  NH1AARG A  63     2344   2778   2472     93    178    -49       N  
ATOM   1004  NH1BARG A  63      13.238  45.076  19.010  0.50 20.32           N  
ANISOU 1004  NH1BARG A  63     2432   2800   2486      5      3      6       N  
ATOM   1005  NH2AARG A  63      13.196  47.250  19.700  0.50 18.22           N  
ANISOU 1005  NH2AARG A  63     2123   2360   2439     95    158    -94       N  
ATOM   1006  NH2BARG A  63      12.180  43.389  17.862  0.50 14.68           N  
ANISOU 1006  NH2BARG A  63     1681   2116   1777    323      6    165       N  
ATOM   1007  H   ARG A  63       6.957  48.858  18.222  1.00  7.57           H  
ATOM   1008  HA  ARG A  63       7.904  47.430  16.129  1.00  7.71           H  
ATOM   1009  HB2AARG A  63       9.063  47.688  18.227  0.50  8.06           H  
ATOM   1010  HB2BARG A  63       8.197  47.083  18.907  0.50  8.61           H  
ATOM   1011  HB3AARG A  63       8.012  46.680  18.837  0.50  8.06           H  
ATOM   1012  HB3BARG A  63       8.594  45.922  17.902  0.50  8.61           H  
ATOM   1013  HG2AARG A  63       8.801  45.096  17.356  0.50 10.37           H  
ATOM   1014  HG2BARG A  63      10.070  47.530  16.882  0.50 11.96           H  
ATOM   1015  HG3AARG A  63       9.984  46.109  16.988  0.50 10.37           H  
ATOM   1016  HG3BARG A  63       9.822  48.411  18.174  0.50 11.96           H  
ATOM   1017  HD2AARG A  63       9.554  45.125  19.620  0.50 13.52           H  
ATOM   1018  HD2BARG A  63      11.805  47.277  18.357  0.50 15.23           H  
ATOM   1019  HD3AARG A  63      10.639  44.498  18.659  0.50 13.52           H  
ATOM   1020  HD3BARG A  63      10.799  46.769  19.479  0.50 15.23           H  
ATOM   1021  HE AARG A  63      10.742  47.189  19.592  0.50 17.17           H  
ATOM   1022  HE BARG A  63      10.697  45.242  17.229  0.50 17.89           H  
ATOM   1023 HH11AARG A  63      12.557  44.604  18.264  0.50 21.28           H  
ATOM   1024 HH11BARG A  63      13.255  45.876  19.321  0.50 21.33           H  
ATOM   1025 HH12AARG A  63      13.874  45.239  18.495  0.50 21.28           H  
ATOM   1026 HH12BARG A  63      13.904  44.553  19.159  0.50 21.33           H  
ATOM   1027 HH21AARG A  63      12.837  47.911  20.108  0.50 21.00           H  
ATOM   1028 HH21BARG A  63      11.516  43.092  17.410  0.50 19.41           H  
ATOM   1029 HH22AARG A  63      14.050  47.224  19.601  0.50 21.00           H  
ATOM   1030 HH22BARG A  63      12.850  42.880  18.008  0.50 19.41           H  
```

### R-value

Regarding model correctness to the experimental data, we can use computer models to model the electron density map of our model. After going through several refinement processes, we can later calculate an $R$ value to determine how similar our model is to the experimental data<a href="#footnote-ref-5" id="footnote-5" className="footnote-ref" aria-describedby="footnote-label" title="Jump to footnote">[5]</a>. The value goes from 0 to 1, with 0 being a perfect match. However, in reality, it's unrealistic to get a perfect match, since the experiment process itself contains a lot of artifacts and noise, matching it to 0 means you will be overfitting to the noise. 

That's why in practice, we use a small holdout set, about 5-10% of the total data gathered. The main data used during the refinement process will produce $R\text{-work}$, while the holdout set will produce $R\text{-free}$. The idea is that if you overfit the data too much to the working set, you will get a bad $R$ value on the holdout set. For a good model, the difference between the two shouldn't be too big, 5% is usually deemed acceptable.

### Steric clashes

Another critical quality check is for **steric clashes**<a href="#footnote-ref-6" id="footnote-6" className="footnote-ref" aria-describedby="footnote-label" title="Jump to footnote">[6]</a>. Think of an atom as having an invisible "personal space" bubble around it. The radius of this bubble is called the van der Waals (vdW) radius. It represents the distance from the atom's nucleus to the point where the attractive forces with a neighboring, non-bonded atom are perfectly balanced by the strong repulsive forces between their electron clouds.

So, why can't this boundary be crossed? The answer lies in the Pauli Exclusion Principle from quantum mechanics, which states that no two electrons can occupy the same quantum state. As two atoms get closer than their combined vdW radii, their electron clouds begin to overlap. This forces electrons into higher energy orbitals, creating an incredibly strong repulsive force that pushes the atoms apart. The energy of this repulsion skyrockets as the distance shortens (it's proportional to $(\frac{1}{r})^{12}$ in the Lennard-Jones potential we'll discuss later). A steric clash in a model represents this physically impossible overlap. High-quality structures will have very few to no steric clashes.

### Ramachandran plot

<div style="text-align: center;  margin: 0 auto; width: 90%">
  ![General Ramachandran plot of 6GLP, visualised with MolProbity](/images/posts/ramachandran_6glp.jpg)
  <p style="font-size: 0.8em; margin-top: -2.5rem;"><b>Figure 10</b>: Ramachandran plot of 6GLP, visualised with MolProbity. <br/> [By Rayhan M, CC BY 4.0]</p>
</div>

Finally, we have the Ramachandran plot<a href="#footnote-ref-7" id="footnote-7" className="footnote-ref" aria-describedby="footnote-label" title="Jump to footnote">[7]</a>. This plot can be used for checking the stereochemical quality of a protein's backbone. It maps the two main dihedral angles of the polypeptide backbone, phi ($\phi$, the C-N-C$\alpha$-C bond) and psi ($\Psi$, the N-C$\alpha$-C-N bond), for every amino acid. The rotation around these bonds is restricted because of steric hindrance between atoms in the backbone and the side chain. The Ramachandran plot visualizes these sterical combinations.

We can split the regions into three categories:

1. **Most Favorable Regions**: These are areas where the backbone atoms are comfortably far apart. A good structure will have >90-95% of its residues here.
2. **Allowed Regions**: Conformations that are less common but still physically possible.
3. **Disallowed Regions**: Conformations where atoms would clash severely. A residue in this region is a major red flag.

The only major exception is Glycine, which lacks a side chain (it just has a hydrogen), giving it far more rotational freedom. Proline is the opposite with its side chain forming a ring with the backbone, making its possible angle far more limited.

<div style="text-align: center;  margin: 0 auto; width: 70%">
  ![Illustration of the electron density map of 2AXW, visualised with COOT.](/images/posts/psi_phi_protein.gif)
  <p style="font-size: 0.8em; margin-top: -2.5rem;"><b>Figure 11</b>: Psi and phi dihedral angles in protein backbone structure. <br/> [By Eric Martz, CC BY-NC-SA 4.0, [YouTube](https://www.youtube.com/watch?v=Q1ftYq13XKk)]</p>
</div>

<div style="text-align: center;  margin: 0 auto; width: 70%">
  ![Illustration of the electron density map of 2AXW, visualised with COOT.](/images/posts/psi_example.gif)
  <p style="font-size: 0.8em; margin-top: -2.5rem;"><b>Figure 12</b>: Psi dihedral angle rotation example in protein backbone. <br/> [By Eric Martz, CC BY-NC-SA 4.0, [YouTube](https://www.youtube.com/watch?v=Q1ftYq13XKk)]</p>
</div>

<div style="text-align: center;  margin: 0 auto; width: 70%">
  ![Illustration of the electron density map of 2AXW, visualised with COOT.](/images/posts/phi_example.gif)
  <p style="font-size: 0.8em; margin-top: -2.5rem;"><b>Figure 13</b>: Phi dihedral angle rotation example in protein backbone. <br/> [By Eric Martz, CC BY-NC-SA 4.0, [YouTube](https://www.youtube.com/watch?v=Q1ftYq13XKk)]</p>
</div>

---

## Storing structural data

To make our model useful and shareable, it must be stored in a standardized, machine-readable format. This is the most boring part of this post, but please bear with me. As knowing how the formats arose, their limitations, and why they still coexist is a neccessity to be able to work with various programs.

### PDB

The Protein Data Bank (PDB) format is the old-timer of structural biology<a href="#footnote-ref-8" id="footnote-8" className="footnote-ref" aria-describedby="footnote-label" title="Jump to footnote">[8]</a>. Developed in the 1970s, it's a fixed-width format with a maximum column of 80. Each line, or "record," has a specific type (like `ATOM` or `HETATM` for non-standard residues) and data is organized into rigid columns. This format stores essential information like atom names, residue names, chain identifiers, XYZ coordinates, and the B-factors we discussed earlier.

However, its age comes with limitations. The PDB format can't handle more than 99,999 atoms or more than 62 chains, making it unsuitable for structures like ribosomes. Moreover, it doesn't explicitly store bond information, so visualisation software has to infer connectivity based on standard residue templates and atom distances, which can sometimes lead to errors.

```shell
HEADER    METHANE EXAMPLE
COMPND    METHANE
AUTHOR    GENERATED EXAMPLE
ATOM      1  C   ACE     1       0.000   0.000   0.000  1.00  0.00           C
ATOM      2  H1  ACE     1       0.629   0.629   0.629  1.00  0.00           H
ATOM      3  H2  ACE     1      -0.629  -0.629   0.629  1.00  0.00           H
ATOM      4  H3  ACE     1      -0.629   0.629  -0.629  1.00  0.00           H
ATOM      5  H4  ACE     1       0.629  -0.629  -0.629  1.00  0.00           H
END
```

### mmCIF

The macromolecular Crystallographic Information File (mmCIF) is the modern, robust successor to the PDB format<a href="#footnote-ref-9" id="footnote-9" className="footnote-ref" aria-describedby="footnote-label" title="Jump to footnote">[9]</a>. It was designed specifically to overcome the limitations of its predecessor. Instead of fixed columns, mmCIF uses a flexible key-value structure, which makes it easily extensible. This format can handle structures of any size and can store a much richer set of metadata, including detailed experimental conditions and quality metrics. As of 2014, the mmCIF format is the official archival standard for the Worldwide Protein Data Bank (wwPDB), ensuring it's the go-to format for modern structural data.

```shell
data_methane
_cell_length_a 5.0
_cell_length_b 5.0
_cell_length_c 5.0
_cell_angle_alpha 90.0
_cell_angle_beta 90.0
_cell_angle_gamma 90.0
_symmetry_space_group_name_H-M 'P 1'

loop_
_atom_site_label
_atom_site_type_symbol
_atom_site_fract_x
_atom_site_fract_y
_atom_site_fract_z
C1 C 0.0000 0.0000 0.0000
H1 H 0.1258 0.1258 0.1258
H2 H -0.1258 -0.1258 0.1258
H3 H -0.1258 0.1258 -0.1258
H4 H 0.1258 -0.1258 -0.1258
```

### SDF

While PDB and mmCIF are designed for macromolecules, the Structure-Data File (SDF) format is the workhorse for small molecules<a href="#footnote-ref-10" id="footnote-10" className="footnote-ref" aria-describedby="footnote-label" title="Jump to footnote">[10]</a>. It's an extension of the simpler MOL format and has a key advantage: a single SDF file can contain thousands or even millions of different molecules, making it ideal for storing chemical libraries. For each molecule, the SDF contains an atom block (with element types and 3D coordinates) and a bond block that explicitly defines which atoms are connected and the order of those bonds. Following the structural information, it can also hold associated data fields for each molecule, such as its name, chemical formula, or experimental activity.

```shell
methane
  -ISIS-  09272513222D

  5  4  0  0  0  0  0  0  0  0999 V2000
    0.0000    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.6290    0.6290    0.6290 H   0  0  0  0  0  0  0  0  0  0  0  0
   -0.6290   -0.6290    0.6290 H   0  0  0  0  0  0  0  0  0  0  0  0
   -0.6290    0.6290   -0.6290 H   0  0  0  0  0  0  0  0  0  0  0  0
    0.6290   -0.6290   -0.6290 H   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  0  0  0  0
  1  3  1  0  0  0  0
  1  4  1  0  0  0  0
  1  5  1  0  0  0  0
M  END
> <ID>
methane_01

> <FORMULA>
CH4

$$$$
```
### XYZ

The XYZ format is on the simpler side of things. It does one thing and one thing only: store atomic coordinates. The format is straightforward: the first line specifies the total number of atoms, the second line is for a comment or title, and every subsequent line lists an element symbol followed by its X, Y, and Z coordinates. That's it. It contains absolutely no information about bonds, charges, residues, or chains. This minimalist approach makes it easy to parse and is widely used in quantum chemistry programs and for simple molecular visualizations where connectivity can be easily inferred. Some programs like ORCA uses .INP which has similar structure to .XYZ.

```shell
5
Methane molecule
C     0.000000    0.000000    0.000000
H     0.629000    0.629000    0.629000
H    -0.629000   -0.629000    0.629000
H    -0.629000    0.629000   -0.629000
H     0.629000   -0.629000   -0.629000
```
### SMILES

Unlike the previous formats, the Simplified Molecular-Input Line-Entry System (SMILES) is not a coordinate file<a href="#footnote-ref-11" id="footnote-11" className="footnote-ref" aria-describedby="footnote-label" title="Jump to footnote">[11]</a>. **It is a line notation**, which allows us to represent a molecule's 2D structure as a 1D string of text. This makes it incredibly compact and perfect for database storage and searching. We can store SMILES as an .smi file.

```shell
# SMILES notation of CH4
C
```

Well, okay, that might not illustrate what SMILES can do, so let's use another example: acetaminophen (N-acetyl-para-aminophenol).

<div style="text-align: center;  margin: 0 auto; width: 60%">
  ![2D structure of acetaminophen](/images/posts/pct.png)
  <p style="font-size: 0.8em; margin-top: -2.5rem;"><b>Figure 14</b>: 2D structure of acetaminophen, generated from the SMILES notation with RDKit. <br/> [By Rayhan M, CC BY 4.0]</p>
</div>

```shell
# SMILES notation of acetaminophen
CC(=O)Nc1ccc(O)cc1
```

While SMILES is a powerful notation, it has a quirk: different algorithms can generate different SMILES strings for the same molecule. This can make it tricky to compare molecules To solve this, "canonical SMILES" algorithms were created to assign a unique SMILES string to each molecule. There are also alternative notations, like [SELFIES](https://arxiv.org/abs/1905.13741) (Self-Referencing Embedded Strings), which are designed to always produce a valid molecule.

### Other formats

There are dozens of other formats that exist in the world of cheminformatics. Some notable examples would be PDBQT (used by Autodock) and PDBQR (used by PDB2PQR). Other softwares like ORCA, GROMACS, and LAMPPS also have their own formats. But need not to worry, you can convert file formats using various existing libraries (e.g., using Meeko for PDBQT or OpenBabel for general purpose conversion). 

```shell
# Examples of file conversions with OpenBabel

obabel input.smi -O output.sdf --gen3d # SMILES to SDF
obabel protein.pdb -O protein.xyz # PDB to XTZ
obabel molecules.sdf -O molecules.smi # SDF to SDF
obabel input.sdf -O output.mol2 # SDF to MOL2
```

# Some refresher on binding

At the heart of many drugs lie a fundamental biological principle: the **interaction between proteins and small molecules**. When proteins go awry, they often become the primary targets for therapeutic intervention.

The core idea for many drugs is quite simple: a small molecule, known as a ligand, binds to a specific site on a target protein, thereby modulating its activity. This binding can inhibit an enzyme essential for a pathogen's survival, block a receptor to halt a pathological signal, or activate a pathway to restore normal function. Of course, there are tons of hand-waving in that explanation, but it will suffice for now. The main goal of structure-based drug design is **designing ligands that are chemically and sterically complementary** to a target protein's binding site. 

<div style="text-align: center;  margin: 0 auto; width: 90%">
  ![Protein can change shape by induced fit upon substrate binding to form a complex.](/images/posts/Hexokinase_induced_fit.svg.png)
  <p style="font-size: 0.8em; margin-top: -2.5rem;"><b>Figure 15</b>: Protein can change shape by induced fit upon substrate binding to form a complex. <br/> [By Thomas Shafee, CC BY 4.0, <a href="https://commons.wikimedia.org/w/index.php?curid=45801557">Wikimedia</a>]</p>
</div>

The **induced fit** model posits that the initial binding event causes conformational changes in both the protein and the ligand to achieve an optimal interaction<a href="#footnote-ref-12" id="footnote-12" className="footnote-ref" aria-describedby="footnote-label" title="Jump to footnote">[12]</a>. An alternative view is the conformational seletion model. That theory suggests that proteins are dynamically changing their conformations, even without a ligand present. In that scenario, the ligand doesn't actively change the protein's shape, but instead "selects" and stabilises a pre-existing conformation that it can bind to. In the modern age, we see those two models as complementary, instead of mutually exclusive. A ligand may first select a favourable conformation from the protein's existing ensemble, then followed by minor mutual adjustment via induced fit to settle the lowest energy state. 

**However, modelling the flexibility of a large molecule like proteins is computationally expensive**, so docking programs usually use the old **lock and key** hypothesis instead. In that hypothesis, we model a rigid protein pocket (the lock) and a perfectly shaped ligand (the key). Yes, it's not perfect, but when you're trying to screen millions of potential leads, you have to make compromises. More advanced techniques like ensemble docking or flexible docking attempt to account for this inherent flexibility, but with a much higher computational cost.

<div style="text-align: center;  margin: 0 auto; width: 90%">
  ![Docking is usually done to "holo" structure](/images/posts/images_large_ci-2018-00730p_0001.jpeg)
  <p style="font-size: 0.8em; margin-top: -2.5rem;"><b>Figure 16</b>: Structural changes of (a, d) BGT, (b, e) RIC, and (c, f) ABP upon binding to their respective ligands UDP, NEO, and ALL. Panels (a-c) highlight the overall conformational shifts of each protein. The apo and holo forms are shown as green and yellow ribbons, respectively. The ligands are represented as stick models coloured by atom type.<br/> J. Chem. Inf. Model. 2019, 59, 4, 1515-1528.</p>
</div>

Another very important caveat that you have to know, is that **docking is usually done to the holo structure of a protein**. What exactly is a holo structure? Well, you see, when a protein binds to other molecule(s), they can undergo conformational changes. In fact, when almost nothing is actively being done to them, proteins still constantly undergo conformational changes! You can see some examples of how apo and holo structures differ in the Figure 13 above.

<div style="text-align: center;  margin: 0 auto; width: 90%">
  ![Docking is usually done to "holo" structure](/images/posts/DHFR_methotrexate_inhibitor.svg.png)
  <p style="font-size: 0.8em; margin-top: -2.5rem;"><b>Figure 17</b>: An enzyme's active site can bind a competitive inhibitor instead of its substrate, blocking access. Methotrexate inhibits dihydrofolate reductase by preventing folic acid binding. Binding site is blue, inhibitor green, substrate black (PDB: 4QI9).<br/> [By Thomas Shafee, CC BY 4.0, <a href="https://commons.wikimedia.org/w/index.php?curid=45802068">Wikimedia</a>]</p>
</div>

Since docking can't simulate proteins conformational changes, people tend to use holo structures, since they already have established binding pockets. Curious minds might wonder: "Won't that limit our candidates to candidates that are quite similar to the native ligand?" If thought so, you're spot on.  This is a major limitation of the conventional docking approach. This inability to model protein flexibility becomes an absolute roadblock when dealing with allostery. Understanding allostery is incredibly complex, with competing theories like the Monod-Wyman-Changeux (MWC) "concerted" model, which proposes an equilibrium between two global protein states, and the Koshland-Némethy-Filmer (KNF) "sequential" model, which describes a step-wise induced fit. Because these mechanisms are fundamentally driven by large-scale, dynamic structural changes, they fall far outside the scope of rigid-protein docking. As such, the aforementioned ensemble docking, which combines MD trajectory and molecular docking to fish pockets from apo structures, is gaining traction to broaden our potential targets.

However, that's beyond our scope here. For now, the only thing you need to know, is that **usually it's better to use the holo structures of whatever protein you're studying, unless you absolutely know what you're doing**. Proteins' binding pocket are usually somewhat flexible, as in, they can accept molecules that are similar enough to the native binder, as can be seen on Figure 14. That fact is what we usually try to exploit when we do virtual screening with docking programs.


# Defining molecular binding

In the previous section, we have discussed about how ligands and proteins interact. However, that understanding is relatively shallow and superficial. We know they bind, but how exactly? How and why they even bind? This section will discuss just that, so that you can understand the factors affecting binding.

## Measuring binding experimentally

Before going deep into the physics of binding, it's important to understand how molecular binding is experimentally measured, as these methods provide the experimental data for validating computational predictions. **Isothermal Titration Calorimetry** (ITC) is widely considered the "gold standard" because it offers a complete thermodynamic profile of molecular interaction in a single, in-solution experiment. ITC directly quantifies the heat released or absorbed during molecular binding. The process involves injecting small aliquots of a ligand from a computer-controlled syringe into a protein solution, maintained at a constant temperature in a sample cell. Each injection generates a heat burst; initially, a large signal indicates abundant free protein binding, which diminishes as binding sites become saturated, eventually reflecting only ligand dilution. The resulting binding isotherm allows for direct determination of binding affinity ($K_A$ or $K_d$), enthalpy ($\DeltaH$), and stoichiometry ($n$). From those metrics, the Gibbs free energy and entropy can then be calculated.

Complementing ITC, **Surface Plasmon Resonance** (SPR) is an optical method designed to measure binding kinetics. In an SPR setup, one binding partner is immobilized on a gold-plated sensor chip. As a solution containing the other molecule flows over the surface, binding increases the mass on the chip. This mass change alters the refractive index, causing a shift in the angle of reflected polarized light, measured in Resonance Units (RU). Subsequently, a buffer without the analyte flows, allowing measurement of the dissociation rate. The resulting sensorgram allows for the extraction of kinetic rate constants: the association rate constant ($k_{on}$ or $k_A$) and the dissociation rate constant ($k_{off}$ or $k_d$).

---

## Binding thermodynamics

Consider binding as an equilibrium of a protein $P$ and a ligand $L$ in an aqueous box of volume V at temperature T, which can be stated as
$$
P + L \rightleftharpoons PL
$$

The rate of $P$ and $L$ binding is a constant of $K_A$ which can be described as
$$
K_A = \frac{k_{\text{on}}}{k_{\text{off}}} 
= \frac{[PL]}{[P][L]} 
= \frac{1}{K_d}
$$

Following that, we can define the molar-standard Gibbs free energy of binding as
$$
\Delta G_{bind} = RTlnK_d = -RTlnK_A
$$

where $K_A$ is the association constant, $K_d$ the dissociation constant, $R$ the gas constant, and $T$ the temperature in kelvin. We can further decompose it to the thermodynamics part:

$$
\Delta G_{bind} = E_{gas} + G_{solv} - \Delta TS_{conf}
$$

where $E_{gas}$ is the interaction of molecules in a vacuum-state, $G_{solv}$ the solvation effect, and $S_{conf}$ the configurational entropy. Now, what are those terms? What do they mean? Don't worry just yet, because they're actually quite intuitive.

$E_{gas}$ and $G_{solv}$ can be categorised into *enthalpy* terms, whereas $S_{conf}$ is the *entropy* term. Let's start with the entropy terms first. **Entropy is the amount of "freedom" lost due to the binding happening**. More concretely, it's usually due to the loss of rotational freedom (rotor) of the molecules. Remember that molecular bonds are not completely rigid, they can flex and rotate. However, when we bind a protein and ligand together, that freedom is gone since they're now interlocked to each other. Since, $G_{bind}$ defines the free energy state of the complex, this reduction to the freedom effectively lessen the entropy of the system, making it an unfavourable interaction.

**Enthalpy is all about the energy of the connections and forces between the protein and the ligand.** It's the sum of the interactions in a vacuum ($E_{gas}$) and the effects of the surrounding environment, which is almost always water ($G_{solv}$) in biological systems. We can further decompose $E_{gas}$ to the sum of its parts:

$$
E_{gas} = E_{bond} + E_{angle} + E_{dihed} + E_{vdw} + E_{coul}
$$

The terms $E_{bond} + E_{angle} + E_{dihed}$ refer to the **internal energy of the molecules**. Imagine the bonds within a molecule as springs. To fit into a protein's binding pocket, a ligand might have to stretch some of its bonds, bend its angles, or twist its dihedral (rotational) angles into a less-than-ideal shape. This distortion creates internal strain, which is an energy penalty. **A good ligand fits comfortably without having to contort itself too much.** However, in practice, we usually ignore these terms since they don't really change much before and after binding (i.e. they cancel out), so we will focus on the non-bonded interactions instead.

### Van der Waals interaction

$E_{vdw}$ or van der Waals is a broad range of interactions: ranging from London dispersion, dipole-dipole, and induced-dipole. Without going into the nitty gritty part, what you have to know is that van der Waals interactions are caused by the movement of electrons. Quantum mechanics posits that the energy of electrons are never zero, that is that they constantly move in an atom through the Schrodinger equation and the Heisenberg uncertainty principle. Hence, at any time, the charge of an atom can be distributed in a different manner. Sometimes electrons are distributed asymmetrically, creating two poles—dipoles—that is one side will be more positive/negative than the other. Usually, many programs will calculate vdW using the Lennard-Jones (LJ) equation<a href="#footnote-ref-13" id="footnote-13" className="footnote-ref" aria-describedby="footnote-label" title="Jump to footnote">[13]</a>:

$$
V_{LJ}(r) = 4\epsilon \left[ \left( \frac{\sigma}{r} \right)^{12} - \left( \frac{\sigma}{r} \right)^6 \right]
$$

Keep in mind that LJ is an approximation, same goes with Morse potential, or any other equations. However, LJ is good enough for most use cases. A negative value means an attraction, while a positive value signifies a repulsion. We can see that there are several parameters in that equations, $\epsilon$ represents the "depth" of the potential well, that is the maximum strength of the attraction between the two particles. $\sigma$ represents the finite distance at which the potential energy is zero. It can be thought of as the effective "size" of a particle or the collision diameter. Lastly, $r$ is the distance between centre of the two particles.

If we take a closer look, we can divide divide the equation into two parts: the repulsive terms $\left( \frac{\sigma}{r} \right)^{12}$ and the attraction terms $\left( \frac{\sigma}{r} \right)^{6}$. Since the repulsive term has a bigger exponent $(^{12})$, it will grow much faster than the attraction terms, this means our vdW energy will quickly become repulsive as molecules get too close, and rapidly approach zero as they move further apart. Moreover, if the distance is shorter than the effective size ($\sigma$), the equation will even produce a positive number, indicating that the particles are too close and repelling each other. I recommend watching this video below by NucleusBiology. Oh also, **hydrogen-bond is a specific type of electrostatic interaction**, if you're wondering.

<YouTube id="x8PpLlKI2yo"
         title="Van der Waals Forces"/>

### Electrostatic interaction

Now, let's continue with the other term: $E_{coul}$. This term basically quantify the effect of electrostatic interaction, which is governed by the Coulomb's law<a href="#footnote-ref-14" id="footnote-14" className="footnote-ref" aria-describedby="footnote-label" title="Jump to footnote">[14]</a>:

$$
F = k\frac{q_1q_2}{r^2}
$$

with $k = 1/4\pi\varepsilon_0$ in a vacuum, $q_1q_2$ the partial charge of the particles, and $r$ the distance between the particles. The value of $\varepsilon_0$ is approximately $\varepsilon_0 = 8.8541878128\times10^{-12}\ \mathrm{F\,m^{-1}}$. If we plug in some numbers, you will notice that the energy produced by this term is *massive*. Let's try calculate an example with a distance of $3 \AA$, $q_1$ of 1, and $q_2$ of -1.

$$
U = \frac{1}{4\cdot\pi\cdot8.8541878128\times10^{-12}}\frac{-1 \cdot 1}{3} = -7.68 \cdot 10^{19} J \approx -110.7\ kcal/mol
$$

Whoops! That's not realistic at all, right? And that's just between two pair of atoms. The reason for this discrepancy is that our calculation was for a vacuum. In a biological system, everything is surrounded by a solvent (usually water in biological systems). The powerful influence of water is captured in the $G_solv$ term, which is arguably the most complex and important factor in molecular binding. We can break it down into two main physical phenomena:

$$
G_{solv} = G_{polar\_solv} + G_{hydrophobic}
$$

### Polar solvation and desolvation penalty

Before binding, any polar or charged groups on both the ligand and the protein's binding pocket are happily interacting with water. Because water is a polar molecule (its oxygen end is slightly negative and its hydrogen ends are slightly positive), it forms stable, ordered "hydration shells" around these charged surfaces.

<div style="text-align: center;  margin: 0 auto; width: 90%">
  ![Docking is usually done to "holo" structure](/images/posts/unhappy_water.jpeg)
  <p style="font-size: 0.8em; margin-top: -2.5rem;"><b>Figure 17</b>: Illustration of the desolvation penalty.<br/> [Courtesy of Francesca Peccati and Gonzalo Jiménez-Osés, Enthalpy-Entropy Compensation in Biomolecular Recognition: A Computational Perspective, ACS Omega 2021 6 (17), 11122-11130, DOI: 10.1021/acsomega.1c00485]</p>
</div>

When the ligand enters the binding pocket, these organized water molecules must be pushed out of the way. This has an energy cost, known as the desolvation penalty, because we have to break the favorable hydrogen bonds between water and the solute molecules. For the binding to be favorable, the new hydrogen bonds and electrostatic interactions formed between the protein and the ligand must be strong enough to overcome this penalty.

This phenomenon also explains why water is so good at screening electrostatic interactions. The hydration shell effectively places a cloud of opposite partial charges around an ion, weakening its interaction with other ions. This screening ability is quantified by the dielectric constant ($\varepsilon_r$). Water has a high dielectric constant of ~80, meaning it reduces the electrostatic force to about 1/80th of its strength in a vacuum. This is why adding $\varepsilon_r$ to the Coulomb's law equation gives a much more realistic energy value:

$$
U = \frac{1}{4\pi\varepsilon_0\varepsilon_r} \frac{q_1q_2}{r}
$$

If we redo our initial calculation with the $\varepsilon_r$ set to 78.5, our $E_{coul}$ becomes **-1.41 kcal/mol**, a much more sensible number! When water is excluded from the binding pocket, the local environment becomes nonpolar (like oil), with a very low dielectric constant ($\varepsilon_r \approx 4$). In this environment, the direct electrostatic attractions that do form between the ligand and protein are much, much stronger.

### The hydrophobic effect

The second, and often dominant, part of solvation is the hydrophobic effect. This is not a direct attraction between oily, nonpolar molecules. Instead, it’s driven by the disorder (entropy) of the surrounding water.

A nonpolar surface disrupts water's natural hydrogen-bonding network. To compensate, the water molecules are forced to form a highly ordered, cage-like structure around the nonpolar group. This ordering represents a significant decrease in entropy, which is thermodynamically unfavorable. The universe prefers chaos!

When two nonpolar surfaces (e.g., a nonpolar part of the ligand and a hydrophobic pocket on the protein) come together, they "hide" from the water which liberates the ordered water molecules from their cages, allowing them to return to the disordered bulk solvent. This release causes a large, favorable increase in the entropy of the system, which provides a powerful thermodynamic push for the binding event.

While simple models like a distance-dependent dielectric can approximate these complex effects, more sophisticated methods are needed for higher accuracy, leading to two primary strategies: explicit and implicit solvation. **Explicit solvation** is the most physically intuitive and accurate approach. It involves building a simulation box where the protein and ligand are surrounded by thousands of individual, explicitly defined water molecules. This method, typically used in molecular dynamics (MD), can capture specific, directed interactions like water-mediated hydrogen bonds.

However, modelling the movement of every single water molecule is computationally expensive, making it impractical for rapid screening. For this reason, molecular docking often rely on **implicit solvation** models. Instead of simulating individual water molecules, it treats the solvent as a continuous, uniform medium with a single dielectric constant, which simplifies the calculation. Other methods like the Poisson-Boltzmann (PB) equation or analytical approximations like the Generalized Born (GB) model are used to calculate the electrostatic contribution of this continuum. While less accurate than explicit models, implicit solvation provides a good-enough approximation of solvent effects at a fraction of the computational cost. For example, the GB equation looks like this:

$$
\Delta G_{\mathrm{elec}}^{\mathrm{GB}}
\;=\; -\dfrac{1}{2}\!\left(1-\dfrac{1}{\varepsilon_s}\right)
\sum_{i=1}^N\sum_{j=1}^N \dfrac{q_i q_j}{f_{ij}}
$$

$$
f_{ij} \;=\; \sqrt{ r_{ij}^2 + R_i R_j \exp\!\Big(-\dfrac{r_{ij}^2}{4 R_i R_j}\Big)\, }
$$

We won't talk too much about the math, but it's important to know that solvation effects, for the most part are heavily simplified in in silico methods due to computational constraints.

### Striking the balance

<div style="text-align: center;  margin: 0 auto; width: 90%">
  ![Docking is usually done to "holo" structure](/images/posts/compensation_entropy_enthalpy.jpeg)
  <p style="font-size: 0.8em; margin-top: -2.5rem;"><b>Figure 17</b>: Entropy and enthalpy cancels out.<br/> [Courtesy of Francesca Peccati and Gonzalo Jiménez-Osés, Enthalpy-Entropy Compensation in Biomolecular Recognition: A Computational Perspective, ACS Omega 2021 6 (17), 11122-11130, DOI: 10.1021/acsomega.1c00485]</p>
</div>

We have talked about both enthalpy and entropy, and how they can be used to calculate the binding affinity of a molecule. However, in reality, the binding affinity is a complex interplay of both. For example, in Figure 17, we can see that the entropy and enthalpy cancels out, any gain in enthalpy is offset by the loss in entropy. As our enthalpy get more favourable, the entropy get less favourable, making the overall binding affinity remain constant. This phenomenon is known as enthalpy-entropy compensation.

To give you a better idea, let's take a look at the binding affinity of a molecule. Let's say we have a complex with a binding affinity of -10 kcal/mol. If we try to refine our ligand by adding a new site for hydrogen bonding, we can perhaps decrease the Gibbs free energy by 1 kcal/mol. However, this hydrogen bond also reduces the entropy of the system by 1 kcal/mol, since any previously free rotamer is now locked in a specific, rigid conformation. As a result, the overall binding affinity remains roughly the same, or sometimes even become less favourable. Due to this phenomenon, it's important to not only consider the enthalpy of the binding, but also the entropy. In the world of drug discovery, we call it as ligand efficiency (LE).


# Conclusion

This brings us to the end of Part 1. We've journeyed from the intricate process of determining a protein's 3D structure to the fundamental thermodynamic forces that govern how a small molecule binds to it. Understanding these concepts--the importance of model quality, the file formats that store structural data, and the enthalpic and entropic contributions to binding energy--lays the essential groundwork for everything that follows. In the next part, we'll peel back the layers of the docking process itself. We'll explore the algorithms that search for the best binding poses and the scoring functions that attempt to predict binding affinity.

Stay tuned!

---

<div className="academic-footnotes">

### References

<span id="footnote-ref-1"><strong>[1]</strong></span> Berman, H. M., et al. (2000). The Protein Data Bank. *Nucleic Acids Research*, 28(1), 235-242. doi:10.1093/nar/28.1.235
<a href="#footnote-1" className="footnote-backref" aria-label="Back to content"> ↩</a>

<span id="footnote-ref-2"><strong>[2]</strong></span> Callaway, E. (2020). The revolution will not be crystallized: a new method sweeps through structural biology. *Nature*, 578(7794), 201-203. doi:10.1038/d41586-020-00341-9
<a href="#footnote-2" className="footnote-backref" aria-label="Back to content"> ↩</a>

<span id="footnote-ref-3"><strong>[3]</strong></span> Rupp, B. (2009). *Biomolecular Crystallography: Principles, Practice, and Application to Structural Biology*. Garland Science.
<a href="#footnote-3" className="footnote-backref" aria-label="Back to content"> ↩</a>

<span id="footnote-ref-4"><strong>[4]</strong></span> Trueblood, K. N., et al. (1996). Atomic displacement parameter nomenclature. Report of a subcommittee on atomic displacement parameter nomenclature. *Acta Crystallographica Section A*, 52(5), 770-781.
<a href="#footnote-4" className="footnote-backref" aria-label="Back to content"> ↩</a>

<span id="footnote-ref-5"><strong>[5]</strong></span> Brünger, A. T. (1992). Free R value: a novel statistical quantity for assessing the accuracy of crystal structures. *Nature*, 355(6359), 472-475.
<a href="#footnote-5" className="footnote-backref" aria-label="Back to content"> ↩</a>

<span id="footnote-ref-6"><strong>[6]</strong></span> Davis, I. W., et al. (2007). MolProbity: all-atom contacts and structure validation for proteins and nucleic acids. *Nucleic Acids Research*, 35(Web Server issue), W375-W383.
<a href="#footnote-6" className="footnote-backref" aria-label="Back to content"> ↩</a>

<span id="footnote-ref-7"><strong>[7]</strong></span> Ramachandran, G. N., Ramakrishnan, C., & Sasisekharan, V. (1963). Stereochemistry of polypeptide chain configurations. *Journal of Molecular Biology*, 7(1), 95-99.
<a href="#footnote-7" className="footnote-backref" aria-label="Back to content"> ↩</a>

<span id="footnote-ref-8"><strong>[8]</strong></span> Protein Data Bank. (1971). Crystallography: protein data bank. *Nature New Biology*, 233(39), 223.
<a href="#footnote-8" className="footnote-backref" aria-label="Back to content"> ↩</a>

<span id="footnote-ref-9"><strong>[9]</strong></span> Westbrook, J. D., et al. (2003). The macromolecular Crystallographic Information File (mmCIF). *Methods in Enzymology*, 374, 394-420.
<a href="#footnote-9" className="footnote-backref" aria-label="Back to content"> ↩</a>

<span id="footnote-ref-10"><strong>[10]</strong></span> Dalby, A., et al. (1992). Description of several chemical structure file formats used by computer programs developed at Molecular Design Limited. *Journal of Chemical Information and Computer Sciences*, 32(3), 244-255.
<a href="#footnote-10" className="footnote-backref" aria-label="Back to content"> ↩</a>

<span id="footnote-ref-11"><strong>[11]</strong></span> Weininger, D. (1988). SMILES, a chemical language and information system. 1. Introduction to methodology and encoding rules. *Journal of Chemical Information and Computer Sciences*, 28(1), 31-36.
<a href="#footnote-11" className="footnote-backref" aria-label="Back to content"> ↩</a>

<span id="footnote-ref-12"><strong>[12]</strong></span> Koshland Jr, D. E. (1958). Application of a theory of enzyme specificity to protein synthesis. *Proceedings of the National Academy of Sciences*, 44(2), 98-104.
<a href="#footnote-12" className="footnote-backref" aria-label="Back to content"> ↩</a>

<span id="footnote-ref-13"><strong>[13]</strong></span> Lennard-Jones, J. E. (1924). On the determination of molecular fields. *Proceedings of the Royal Society of London. Series A*, 106(738), 463-477.
<a href="#footnote-13" className="footnote-backref" aria-label="Back to content"> ↩</a>

<span id="footnote-ref-14"><strong>[14]</strong></span> Coulomb, C. A. (1785). Premier mémoire sur l'électricité et le magnétisme. *Histoire de l'Académie Royale des Sciences*, 569-577.
<a href="#footnote-14" className="footnote-backref" aria-label="Back to content"> ↩</a>

<span id="footnote-ref-15"><strong>[15]</strong></span> Freire, E., Mayorga, O. L., & Straume, M. (1990). Isothermal titration calorimetry. *Analytical Chemistry*, 62(18), 950A-959A.
<a href="#footnote-15" className="footnote-backref" aria-label="Back to content"> ↩</a>

<span id="footnote-ref-16"><strong>[16]</strong></span> Karlsson, R., et al. (1991). Real-time biospecific interaction analysis using surface plasmon resonance and a sensor chip technology. *BioTechniques*, 11(5), 620-627.
<a href="#footnote-16" className="footnote-backref" aria-label="Back to content"> ↩</a>

### Acknowledgments

Special thanks to the Protein Data Bank and the structural biology community for making structural data freely available, enabling research and education worldwide.

### Author Contributions

This article was researched, written, and developed by the author. All figures were created or adapted with appropriate attribution to their original sources.

### License

Diagrams and text are licensed under Creative Commons Attribution CC-BY 4.0 with the source available on GitHub, unless noted otherwise.

</div>
